soco/music_services/music_service.py

```diff
@@ -0,0 +1,780 @@
+    #    getExtendedMetadataText(xs:string id, xs:string Type)
+    #    getLastUpdate()
+    #    getMediaMetadata(xs:string id)
+    #    getMediaURI(xs:string id)
+    #    getMetadata(xs:string id, xs:int index, xs:int count,xs:boolean
+    #                recursive)
+    #    getScrollIndices(xs:string id)
+    #    getSessionId(xs:string username, xs:string password)
+    #    mergeTrialccount(xs:string deviceId)
+    #    rateItem(id id, xs:integer rating)
+    #    search(xs:string id, xs:string term, xs:string index, xs:int count)
+    #    setPlayedSeconds(id id, xs:int seconds)
+
+    def get_metadata(
+            self, item_id='root', index=0, count=100, recursive=False):
+        """Get metadata for a container or item.
+
+        Args:
+            item_id (str): The container or item to browse. Defaults to the
+                root item
+            index (int): The starting index. Default 0
+            count (int): The maximum number of items to return. Default 100
+            recursive (bool): Whether the browse should recurse into sub-items
+                (Does not always work). Defaults to False
+        Returns:
+            (OrderedDict): The item or container's metadata, or None.
+                See http://musicpartners.sonos.com/node/83
+
+        """
+        response = self.soap_client.call(
+            'getMetadata', [
+                ('id', item_id),
+                ('index', index), ('count', count),
+                ('recursive', 1 if recursive else 0)]
+        )
+        return response.get('getMetadataResult', None)
+
+    def search(self, category, term='', index=0, count=100):
+        """Search for an item in a category.
+
+        Args:
+            category (str): The search category to use. Standard Sonos search
+                categories are 'artists', 'albums', 'tracks', 'playlists',
+                'genres', 'stations', 'tags'. Not all are available for each
+                music service. Call available_search_categories for a list for
+                this service.
+            term (str): The term to search for
+            index (int): The starting index. Default 0
+            count (int): The maximum number of items to return. Default 100
+
+        Returns:
+            (OrderedDict): The search results, or None
+                See http://musicpartners.sonos.com/node/86
+
+        """
+        search_category = self._get_search_prefix_map().get(category, None)
+        if search_category is None:
+            raise MusicServiceException(
+                "%s does not support the '%s' search category" % (
+                    self.service_name, category))
+
+        response = self.soap_client.call(
+            'search',
+            [
+                ('id', search_category), ('term', term), ('index', index),
+                ('count', count)])
+        return response.get('searchResult', None)
+
+    def get_media_metadata(self, item_id):
+        """Get metadata for a media item.
```

Comment:
This is sort of what I meant. This return "metadata" for an (media) item. But the ``get_metadata`` gets metadata for an item or a container. Is it the same, if it is for an item?